### DISCUSSION TITLE
`css` props are not applied within the server component.

### DISCUSSION THREAD

**Discussion by wjdgus09** [styled-components/styled-components#4277] [Q&A]
### Environment
- next.js version: 13.4.9
- styled-components version: 6.1.1

### Description
I'm developing a design system package (this is a react client component package) that uses styled-components internally.

For example,

```javascript
# deploy `MyComponent` at npm registry

import styled from 'styled-components'

export const MyComponent = styled.div`
	/** ... */
`
```

and use like this

```javascript
import { MyComponent } from '@test/my-component`

export function ServerComponent() {
	// ...
	return (
		<MyComponent css={{ margin: 10 }} />
	)
}
```

I confirmed that [`css prop`](https://styled-components.com/docs/api#css-prop) is not applied when the component of my package (client component) is imported and used within the server component.

`next.config.js` settings for using styled-component have been applied to the project using the package.

```javascript
  compiler: {
    styledComponents: true,
  },
```

I wonder if this was intentional.
Is css-prop not applicable within the server component?

**Comment by wjdgus09** [comment]
move to issue

**Comment by quantizor** [comment]
The upcoming release includes significant RSC improvements — styled components now work directly in React Server Components via inline `<style>` tag emission, without needing `'use client'` for basic styling.

You can test the prerelease now:

```
npm install styled-components@test
```

The staged release PR is at https://github.com/styled-components/styled-components/pull/5687. If you're still experiencing issues with the `css` prop in server components after upgrading, please report back here.
